lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt

```diff
@@ -19,6 +20,7 @@ import kotlinx.serialization.ExperimentalSerializationApi
 import kotlinx.serialization.Serializable
 import kotlinx.serialization.SerializationException
 import kotlinx.serialization.json.*
+import java.lang.IllegalArgumentException
 import java.util.function.Function
```

Comment:
Let's not use exceptions from java.lang, but directly from kotlin.

Reply:
Changed to `kotlin.IllegalArgumentException`